packages/embark-ui/src/services/web3.js

```diff
@@ -21,13 +21,12 @@ export function estimateGas({web3, contract, args}) {
 
 export function deploy({web3, contract, args}) {
   return new Promise((resolve, reject) => {
-    const cleanup = () => { promiEvent.removeAllListeners(); };
-    const promiEvent = new web3.eth.Contract(contract.abiDefinition)
+    new web3.eth.Contract(contract.abiDefinition)
       .deploy({data: `0x${contract.code}`, arguments: args})
       .send({from: web3.eth.defaultAccount})
       .on('error', reject)
```

Comment:
```suggestion
      .once('error', reject)
```

Reply:
Also, I thought about it a little more since yesterday, and I since we're doing `.catch(reject)` we don't need this event handler.  I suggest we remove it, and then if things don't seem to be working right (perhaps owing to a bug/s in web3's combo Promise-EventEmitter class), we can flip to `.catch(() => {})` and `.once('error', reject)`.

Reply:
See: https://github.com/embark-framework/embark/pull/1360#discussion_r261301208.